### DISCUSSION TITLE
How to fix multiple punctuation mistakes in one quick fix?

### DISCUSSION THREAD

**Discussion by ghost** [valentjn/vscode-ltex#180] [Q&A] [answered]
Especially those punctuations that cannot produce from keyboard or been converted to other format such as smart quotes and m-dash.

I already read on [how to fix multiple spelling errors at the same time,](https://valentjn.github.io/vscode-ltex/docs/faq.html#how-can-i-fix-multiple-spelling-errors-at-the-same-time) but I still don't know what to do.

Sometime, I catch incomplete quotes, but LTeX doesn't. Is there a way to make LTeX catch these mistake?

I often use `Pandoc` to convert `DOCX`, `ODT`, and `HTML` to `MD`, but it doesn't bring the punctuation (It also brings backslash in weird place). Is there a way to prevent this?

I hope you don't mind of these question, thanks.

**Comment by valentjn** [comment, marked as answer]
Regarding multiple fixes at once: Just as the FAQ answer says, select all the text, press `Ctrl` and `.` (period) together (you can instead just click on the light bulb), and then select the fix you want to apply (either via keyboard or mouse).

![vscode-ltex-180-1](https://user-images.githubusercontent.com/19839841/100546227-78363600-3260-11eb-82f4-90075c743e63.gif)

Regarding incomplete quotes: LT<sub>E</sub>X is basically a frontend for [LanguageTool (LT)](https://github.com/languagetool-org/languagetool), which does the actual checking. There is an LT rule `EN_UNPAIRED_BRACKETS` that checks for unpaired quotation marks, but it doesn't seem to be well-implemented (e.g., it doesn't work for "smart quotes", and it produced false positives, see languagetool-org/languagetool#2356). The rule doesn't work at least for me, unfortunately.

Regarding Pandoc: LT<sub>E</sub>X can't influence what the output of Pandoc is. If LT<sub>E</sub>X displays punctuation errors for Markdown files generated by Pandoc, either fix all the punctuation errors with a single quick fix as shown above, or disable the punctuation rule(s) by selecting `Disable rule` after pressing `Ctrl` and `.` (period). If you think the output of Pandoc is wrong (e.g., the backslashes you mentioned), you have to ask the people at [jgm/pandoc](https://github.com/jgm/pandoc).

As a side note: I thought about integrating Pandoc support into LT<sub>E</sub>X to be able to spell-check formats like HTML directly without converting them to Markdown, but this is currently not possible due to technical reasons, see #91.

**Comment by ghost** [comment]
Wow, this answer is better than I expected, thanks.
